includes/sanitizers/class-amp-nav-menu-dropdown-sanitizer.php

```diff
@@ -0,0 +1,164 @@
+<?php
+/**
+ * Class AMP_Nav_Menu_Dropdown_Sanitizer
+ *
+ * @package AMP
+ */
+
+/**
+ * Class AMP_Nav_Menu_Dropdown_Sanitizer
+ *
+ * Handles state for navigation menu dropdown toggles, based on theme support.
+ *
+ * @since 1.1.0
+ */
+class AMP_Nav_Menu_Dropdown_Sanitizer extends AMP_Base_Sanitizer {
+
+	/**
+	 * Default args.
+	 *
+	 * @since 1.1.0
+	 * @var array
+	 */
+	protected $DEFAULT_ARGS = array(
+		'sub_menu_button_class'        => '',
+		'sub_menu_button_toggle_class' => '',
+		'expand_text'                  => '',
+		'collapse_text'                => '',
+		'icon'                         => null, // Optional.
+		'sub_menu_item_state_id'       => 'navMenuItemExpanded',
+	);
+
+	/**
+	 * AMP_Nav_Menu_Dropdown_Sanitizer constructor.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @param DOMDocument $dom  DOM.
+	 * @param array       $args Args.
+	 */
+	public function __construct( $dom, $args = array() ) {
+		parent::__construct( $dom, $args );
+
+		$this->args = self::ensure_defaults( $this->args );
+	}
+
+	/**
+	 * Add filter to manipulate output during output buffering to add AMP-compatible dropdown toggles.
+	 *
+	 * @since 1.0
+	 *
+	 * @param array $args Args.
+	 */
+	public static function add_buffering_hooks( $args = array() ) {
+		$theme_support_args = AMP_Theme_Support::get_theme_support_args();
```

Comment:
Note that the concern below about calling `AMP_Theme_Support::get_theme_support_args()` inside of the `sanitize` method does not apply here for `add_buffering_hooks`. 

Nevertheless, should it not only be looking at `$args`? The `$args` is coming from the `amp_get_content_sanitizers()` function from an applied `amp_content_sanitizers` filter. So either the theme support args could be passed into that array at that point (as you're already doing above). So is this not redundant here anyway?